File: fsharp/hanoi_tower.fs
let rec hanoi n, a, b, c = function
    | 1 -> 1
    | n -> hanoi (n - 1, a, b, c) + 1 + hanoi (n - 1, b, c, a)

let value = hanoi(32, 'a', 'b', 'c')
printfn "Moves: %d" value

(* not checked, fsharp does not work on my system xD *)

File: ruby/hanoi_tower.rb
a = Object.new.freeze
b = Object.new.freeze
c = Object.new.freeze

def hanoi(n, from, mid, to)
  return 1 if n == 1

  hanoi(n - 1, from, to, mid) + 1 + hanoi(n - 1, mid, from, to)
end

result = hanoi(32, a, b, c)
puts "#{result} moves"
